plugins/rtmp-services/rtmp-common.c

```diff
@@ -478,6 +482,32 @@ static void fill_stream_key_link(json_t *service, obs_data_t *settings)
 				    stream_key_link);
 }
 
+static void update_protocol(json_t *service, obs_data_t *settings)
+{
+	const char *protocol = get_string_val(service, "protocol");
+	if (protocol) {
+		obs_data_set_string(settings, "protocol", protocol);
+		return;
+	}
+
+	json_t *servers, *server;
+	const char *url;
+	size_t idx = 0;
+	char *prefix = NULL;
+	servers = json_object_get(service, "servers");
+	if (!json_is_array(servers))
+		return;
+	server = json_array_get(servers, 0);
+	url = get_string_val(server, "url");
+
+	if (strncmp(url, RTMPS_PREFIX, strlen(RTMPS_PREFIX)) == 0) {
+		obs_data_set_string(settings, "protocol", "RTMPS");
+		return;
+	}
+
+	obs_data_set_string(settings, "protocol", "RTMP");
+}
```

Comment:
The use `obs_data_set_*` is necessary for now, since protocol selection is a thing only in RFC 39.
We need to manage the protocol in this service this way for now.

Reply:
Unresolve it if it needs more discussion.